pkg/controllers/apply_controller.go

```diff
@@ -316,15 +331,6 @@ func mergeOwnerReference(owners, newOwners []metav1.OwnerReference) []metav1.Own
 	return owners
 }
 
-func findWorkReference(owners []metav1.OwnerReference, workOwner metav1.OwnerReference) bool {
```

Comment:
why remove this function when line 203 is just a repeat of this logic?

Reply:
I have brought this function back. (Named to findOwnerReference)
I noticed that the pre-existing insertOwner did not include the UID in the check, I have included that in the findOwnerReference.
Also, within the applyUnstructured where findOwnerReference is being used, 
I have used WorkObj.GetOwnerReference()[0], since there are most likely just 1 owner, looking at how we create the WorkObj.